extensions/interactions/DragAndDropSortInput/DragAndDropSortInput_test.py

```diff
@@ -0,0 +1,193 @@
+                        'html': '<p>Choice 2</p>'
+                    }]
+                }
+            },
+            'default_outcome': {
+                'dest': 'abc',
+                'feedback': {
+                    'content_id': 'feedback_1',
+                    'html': 'Correct Answer'
+                },
+                'param_changes': [],
+                'refresher_exploration_id': None,
+                'labelled_as_correct': True,
+                'missing_prerequisite_skill_id': None
+            },
+            'hints': [{
+                'hint_content': {
+                    'content_id': 'hint_1',
+                    'html': '<p>This is a copyright character ©.</p>'
+                }
+            }],
+            'solution': {
+                'answer_is_exclusive': True,
+                'correct_answer': [['<p>Choice 1</p>', '<p>Choice 2</p>']],
+                'explanation': {
+                    'content_id': 'solution',
+                    'html': 'This is <i>solution</i> for state1'
+                }
+            },
+            'id': 'DragAndDropSortInput'
+        }
+        drag_and_drop_instance = (
+            interaction_registry.Registry.get_interaction_by_id(
+                'DragAndDropSortInput'))
+        interaction_domain = (
+            state_domain.InteractionInstance.from_dict(
+                interaction_dict))
+        drag_and_drop_proto = (
+            drag_and_drop_instance.to_android_drag_drop_input_proto(
+                interaction_domain.default_outcome,
+                interaction_domain.customization_args,
+                interaction_domain.solution,
+                interaction_domain.hints,
+                interaction_domain.answer_groups))
+
+        self.assertEqual(
+            drag_and_drop_proto.default_outcome.destination_state,
+            'abc')
+        self.assertEqual(
+            drag_and_drop_proto.default_outcome.feedback.content_id,
+            'feedback_1')
+        self.assertEqual(
+            drag_and_drop_proto.default_outcome.feedback.text,
+            'Correct Answer')
+        self.assertTrue(drag_and_drop_proto.default_outcome.labelled_as_correct)
+
+        self.assertEqual(
+            drag_and_drop_proto.hints[0].hint_content.content_id,
+            'hint_1')
+        self.assertEqual(
+            drag_and_drop_proto.hints[0].hint_content.text,
+            '<p>This is a copyright character ©.</p>')
+
+        self.assertEqual(
+            drag_and_drop_proto.solution.base_solution.explanation.content_id,
+            'solution')
+        self.assertEqual(
+            drag_and_drop_proto.solution.base_solution.explanation.text,
+            'This is <i>solution</i> for state1')
+
+        self.assertEqual(
+            drag_and_drop_proto.solution.correct_answer.content_id_sets[0]
+                .content_ids[0].content_id,
+            '<p>Choice 1</p>')
+
+        drag_input_rule_spec = (
```

Comment:
Please use the full name for the interaction, even in local variables. Ditto elsewhere.

Reply:
Fixed all, with proper name.